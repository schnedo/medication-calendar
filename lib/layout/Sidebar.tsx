import { ReactElement, ReactEventHandler } from "react";
import {
  SwipeableDrawer,
  Link as MuiLink,
  List,
  makeStyles,
  ListItem,
  ListItemText,
} from "@material-ui/core";
import Link from "next/link";
import { title as userInfoTitle } from "../../pages/userInfo";
import { title as doctorInfoTitle } from "../../pages/doctorInfo";

const useStyles = makeStyles((theme) => ({
  drawer: {
    "&>*": { minWidth: theme.spacing(20) },
  },
}));

interface SidebarLinkProps {
  href: string | URL;
  text: string;
}
function SidebarLink({ href, text }: SidebarLinkProps) {
  return (
    <ListItem button>
      <ListItemText>
        <Link href={href} passHref>
          <MuiLink color={"inherit"} underline={"none"}>
            {text}
          </MuiLink>
        </Link>
      </ListItemText>
    </ListItem>
  );
}

export interface SidebarProps {
  open: boolean;
  onOpen: ReactEventHandler;
  onClose: ReactEventHandler;
}

export default function Sidebar({
  open,
  onOpen,
  onClose,
}: SidebarProps): ReactElement {
  const { drawer } = useStyles();
  return (
    <SwipeableDrawer
      className={drawer}
      open={open}
      onOpen={onOpen}
      onClose={onClose}
    >
      <List>
        <SidebarLink href={"/userInfo"} text={userInfoTitle} />
        <SidebarLink href={"/doctorInfo"} text={doctorInfoTitle} />
      </List>
    </SwipeableDrawer>
  );
}
