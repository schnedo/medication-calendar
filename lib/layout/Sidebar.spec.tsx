import { render } from "@testing-library/react";
import Sidebar from "./Sidebar";

describe("Sidebar", () => {
  it("should render correctly", async () => {
    expect.hasAssertions();

    const { container } = render(
      <Sidebar open={true} onOpen={jest.fn()} onClose={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
