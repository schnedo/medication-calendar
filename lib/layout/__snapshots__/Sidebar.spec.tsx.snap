// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="PrivateSwipeArea-root-2 PrivateSwipeArea-anchorLeft-3"
    style="width: 20px;"
  />
</div>
`;
